README.md

```diff
@@ -45,58 +45,140 @@ This project can be utilized one of two ways, via docker or via a Makefile inclu
+    ```
+
+    The target `install/gh-release/%` as well as the vars `$(OS)` and `$(ARCH)` are
+    provided by the tardigrade-ci/Makefile, and do not need to be redefined in your
+    local Makefile.
+
+## Makefile prerequisites
+
+Using the Makefile approach involves calling `make` directly. This means make,
+as well as a few other prerequisites must be installed on the system for it to
+work.
+
+* `make` v4.2 or later
+* `jq` v1.5 or later
+* `git` v2.20 or later
+* `bash` v4 or later
+* `sed` v4 or later
+* `awk` v4 or later
+* `grep` v3 or later
+* `xargs` v4.7 or later
+* `curl` v7 or later
+
+The Makefile is written with Debian-based systems in mind, including Ubuntu, and
+all those packages can be installed using `apt` from default repos. Other platforms
+ought to work fine, as long as these packages/versions can be installed. For example,
+MacOS is known to work, using `brew` to get a modern version of `bash` and GNU
+tools (there are many guides, [here is one](https://www.topbug.net/blog/2013/04/14/install-and-use-gnu-command-line-tools-in-mac-os-x/)).
+
+There are a couple make targets that call out to `apt` directly. You can avoid
+that requirement by pre-installing these packages using your own package manager,
+or simply do not invoke targets that rely on these packages:
+
+* `zip`
+* `nodejs`
+
+## Makefile "operating modes"
+
+When using the Makefile approach, the `include` directive is a shell command that
+retrieves the "bootstrap" Makefile from this project, saves it to the current working
+directory, and then echoes back the name of the saved file. Make reads the bootstrap
+Makefile and processes its directives and targets.
+
+The bootstrap Makefile is very barebones. Its primary purpose is to bootstrap this
+`tardigrade-ci` project so all the make targets are available. It does this through
+its own `include` directive. The bootstrap include directive processes a shell
+command with logic to determine whether to clone this project, and then echoes
+back the path to this project's Makefile.
+
+The bootstrap Makefile supports two "operating modes" that we're calling **user mode**
+and **developer mode**.
+
+In **user mode**, the bootstrap Makefile automatically clones the `tardigrade-ci`
+repo to a subdirectory of the calling project, e.g. `/{project}/tardigrade-ci/`.
+This feature is called **auto-init**. In this mode, the `tardigrade-ci` subdirectory
+is considered "owned" by the bootstrap Makefile. The bootstrap Makefile will
+manage and update the clone (through the `include` logic) to ensure the checkout
+matches the git branch or tag specified by the env `TARDIGRADE_CI_BRANCH`. By
+default, this env will resolve to either the version specified in the `Dockerfile`
+(if present) or the `master` branch (if the Dockerfile is missing or is not
+specifying a version). The user may override the default behavior by providing
+the env explicitly, e.g. `TARDIGRADE_CI_BRANCH=foo make help`.
```

Comment:
The example value for TARDIGRADE_CI_BRANCH is confusing.  The value is to be a branch, not a command, or am I missing the point?

Reply:
`foo` was meant to indicate the branch value. How about `{branch}` instead?

```suggestion
the env explicitly, e.g. `TARDIGRADE_CI_BRANCH={branch} make help`.
```

Reply:
I'm still confused ... is TARDIGRADE_CI_BRANCH a branch name?  If so, why are you including a command with it?

Reply:
`TARDIGRADE_CI_BRANCH` is the env that is used by the bootstrap makefile to set the branch (or tag) for the `init` functionality.

`TARDIGRADE_CI_BRANCH={branch} make help` is shell syntax for setting an env for a single command, without needing to export it into the shell.

Reply:
That's obvious now, but I couldn't see it before.  I'll look at the text again on Monday as this was confusing.

Reply:
The earlier part of the paragraph references the envvar TARDIGRADE_CI_BRANCH, so I don't know why I would think that you'd need to just show how to set that envvar.  It is dense reading and you can get lost in the weeds, so maybe change:

```
The user may override the default behavior by providing the env explicitly, e.g., ...
```

with

```
The user may override the default behavior by providing the env explicitly before invoking make, e.g., ...
``` 

It's a bit lame, but I kept missing the obvious.

Reply:
Roger. Perhaps I'm trying too hard. How about we just remove that sentence? If we need to, we can add a separate section on the bootstrap envs.

Reply:
I would keep the sentence and add the "before invoking make".  

You have a lot of envvars.  Most people won't need to know that level of detail, i.e., all the envvars.  @eemperor - what do you think?

Reply:
@kbalk I've updated the wording.

Reply:
Thank you.